tasks/image/build.go

```diff
@@ -69,7 +69,17 @@ func buildIsStale(ctx *context.ExecuteContext, t *Task) (bool, error) {
 	if t.config.Steps != "" && ctx.ConfigFile != "" {
 		paths = append(paths, ctx.ConfigFile)
 	}
-	mtime, err := fs.LastModified(paths...)
+
+	excludes := []string{}
+	dockerignoreEntries, err := build.ReadDockerignore(t.config.Context)
+	if err == nil {
+		excludes = dockerignoreEntries
+	} else {
+		t.logger().Warnf("Failed to read .dockerignore file.")
+	}
```

Comment:
It is safe to append to a `nil` slice (https://play.golang.org/p/BuzI0k5swDn), so I think this logic can be simplified a bit:

```go
excludes, err := build.ReadDockerignore(t.config.Context)
if err != nil {
	t.logger().Warnf("Failed to read .dockerignore file.")
}
```